src/array.vala

```diff
@@ -2,6 +2,9 @@ using GLib;
 
 public class Vast.Array : Object
 {
+    public const ssize_t NEW_AXIS = ssize_t.MAX;
+    public const size_t DEFAULT_DIMENSION = size_t.MAX;
```

Comment:
Maybe we could have a view that extends the dimension instead of having a default dimension?

```vala
var arr = new Vast.Array (..., {2, 2, 2, 2});
arr.redim (5).build ().slice (4, 0, 1);
```

That would basically call `reshape` with paddings of `1`.